modzy/_util.py

```diff
@@ -32,3 +34,44 @@ def file_to_bytes(file_like):
 
     with open(path, 'rb') as file:
         return file.read()
+
+
+def file_to_chunks(file_like, chunk_size):
+    logger.debug(f"file_to_chunks({type(file_like)} :: {file_like}, {chunk_size}) :: splitting ")
+    file = None
+    if not hasattr(file_like, 'read'):
+        if hasattr(file_like, '__fspath__'):  # os.PathLike
+            path = file_like.__fspath__()
+        elif isinstance(file_like, pathlib.Path):  # Python 3.4-3.5
+            path = str(file_like)
+        else:
+            path = file_like
+        file = open(path, 'rb')
+    else:
+        file = file_like
+
+    if hasattr(file, 'seekable') and file.seekable():
+        file.seek(0)
+
+    i = 0
+    while True:
+        chunk = file.read(chunk_size)
+        logger.debug(f"file_to_chunks({type(file)}, {chunk_size}) :: chunk [{i}:{i + chunk_size}]")
+        i += 1
+        if not chunk:
+            break
+        if not isinstance(chunk, bytes):
+            raise TypeError("the file object's 'read' function must return bytes not {}; "
+                            "files should be opened using binary mode 'rb'"
+                            .format(type(chunk).__name__))
+        yield chunk
+
+    if file and hasattr(file, 'close'):
```

Comment:
This closing behavior as the final pass over an iterator seems unusual as the user might expect to see an object or get a value returned. Maybe using a with statement and/or a context manager here could be useful